src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java

```diff
@@ -0,0 +1,131 @@
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.webserver.api.passwordless;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.supertokens.Main;
+import io.supertokens.passwordless.Passwordless;
+import io.supertokens.passwordless.Passwordless.FieldUpdate;
+import io.supertokens.pluginInterface.RECIPE_ID;
+import io.supertokens.pluginInterface.emailpassword.exceptions.DuplicateEmailException;
+import io.supertokens.pluginInterface.emailpassword.exceptions.UnknownUserIdException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.passwordless.UserInfo;
+import io.supertokens.pluginInterface.passwordless.exception.DuplicatePhoneNumberException;
+import io.supertokens.utils.Utils;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class UserAPI extends WebserverAPI {
+
+    private static final long serialVersionUID = -4641988458637882374L;
+
+    public UserAPI(Main main) {
+        super(main, RECIPE_ID.PASSWORDLESS.toString());
+    }
+
+    @Override
+    public String getPath() {
+        return "/recipe/user";
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        // logic based on: https://app.code2flow.com/flowcharts/617a9aafdc97ee415448db74
+        String userId = InputParser.getQueryParamOrThrowError(req, "userId", true);
+        String email = InputParser.getQueryParamOrThrowError(req, "email", true);
+        String phoneNumber = InputParser.getQueryParamOrThrowError(req, "phoneNumber", true);
+
+        if (Stream.of(userId, email, phoneNumber).filter(Objects::nonNull).count() != 1) {
+            throw new ServletException(
+                    new BadRequestException("Please provide exactly one of userId, email or phoneNumber"));
+        }
+
+        try {
+            UserInfo user;
+            if (userId != null) {
+                user = Passwordless.getUserById(main, userId);
+            } else if (email != null) {
+                user = Passwordless.getUserByEmail(main, Utils.normaliseEmail(email));
+            } else {
+                user = Passwordless.getUserByPhoneNumber(main, phoneNumber);
+            }
+
+            if (user == null) {
+                JsonObject result = new JsonObject();
+                result.addProperty("status", userId != null ? "UNKNOWN_USER_ID_ERROR"
+                        : email != null ? "UNKNOWN_EMAIL_ERROR" : "UNKNOWN_PHONE_NUMBER_ERROR");
```

Comment:
Add brackets like `userId != null ? "UNKNOWN_USER_ID_ERROR" : (email != null ? "UNKNOWN_EMAIL_ERROR" : "UNKNOWN_PHONE_NUMBER_ERROR")` just to make this super clear